pydantic/class_validators.py

```diff
@@ -147,7 +147,11 @@ def _prepare_validator(function: AnyCallable, allow_reuse: bool) -> 'AnyClassMet
     """
     f_cls = function if isinstance(function, classmethod) else classmethod(function)
     if not in_ipython() and not allow_reuse:
-        ref = f_cls.__func__.__module__ + '.' + f_cls.__func__.__qualname__
+        ref = (
+            getattr(f_cls.__func__, '__module__', '<default_module>')
+            + '.'
+            + getattr(f_cls.__func__, '__qualname__', '<default name>')
```

Comment:
Because uniqueness is required, and `__qualname__` was used here in part to provide uniqueness, I would replace `<default_name>` with something like `f'id:{id(f_cls.__func__)}'`

Reply:
TBH: I think less uniqueness is not that bad, when we are checking for uniqueness; at worst it will mark a function as duplicate that isn't; this should prevent the problem this check has been created for

Reply:
The problem is it would raise an error any time you use two or more partials as validators, regardless of their relationship, unless you modify the model config to allow that. I don't have a particularly strong opinion on this, but I think using the `id` is more representative of current behavior, as things with a `__qualname__` will generally have the _same_ `__qualname__` if and only if they have the same `id`. (Maybe there are exceptions to that, but if there are, it's probably not "normal" usage.)

Reply:
That's the reason I asked about allowing those only with `allow_reuse`.

~TBH I worry more on the other case~ That makes no sense as we are only using the id as fallback